zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -311,16 +312,19 @@ private boolean registerInterpreterFromResource(ClassLoader cl, String interpret
     URL[] urls = recursiveBuildLibList(new File(interpreterDir));
     ClassLoader tempClassLoader = new URLClassLoader(urls, cl);
 
-    InputStream inputStream = tempClassLoader.getResourceAsStream(interpreterJson);
-
-    if (null != inputStream) {
-      logger.debug("Reading {} from resources in {}", interpreterJson, interpreterDir);
-      List<RegisteredInterpreter> registeredInterpreterList =
-          getInterpreterListFromJson(inputStream);
-      registerInterpreters(registeredInterpreterList, interpreterDir);
-      return true;
+    Enumeration<URL> interpreterSettings = tempClassLoader.getResources(interpreterJson);
+    if (!interpreterSettings.hasMoreElements()) {
+      return false;
     }
-    return false;
+    for (URL url : Collections.list(interpreterSettings)) {
+      try (InputStream inputStream = url.openStream()) {
+        logger.debug("Reading {} from {}", interpreterJson, url);
+        List<RegisteredInterpreter> registeredInterpreterList =
+            getInterpreterListFromJson(inputStream);
+        registerInterpreters(registeredInterpreterList, interpreterDir);
```

Comment:
I think it is very hard to control when we allow multiple interpreter-setting.json. It would bring several potential problems. like which interpreter-setting.json will be read first ? Will the later one overwrite the previous one ? All these problems depends on the implementation.  BTW, what is the exact problem of this PR try to fix ? It seems need to create a ticket for it.